docs/development/plugins/publishing.md

```diff
@@ -0,0 +1,100 @@
+---
+title: Publishing Plugins
+sidebar_label: Publishing
+---
+
+Headlamp in its core does not have a way to make plugins discoverable or to list them to the user. However, the desktop version of Headlamp ships with the [Plugin Catalog](URL_FOR_PLUGIN) plugin which does provide a way for users to list Headlamp plugins from [ArtifactHub](https://artifacthub.io), install and remove them.
+
+This section of the docs is a guide for how to publish a Headlamp plugin to ArtifactHub.
+
+## Prerequisites
+
+* A GitHub repository containing your Headlamp plugin code. See the [building](./building.md) section to learn how to implement a Headlamp plugin.
+* The [headlamp-plugin](https://www.npmjs.com/package/@kinvolk/headlamp-plugin) NPM package version 0.10.0 (or higher) installed: you can do this by running `npm install -g @kinvolk/headlamp-plugin@^0.10`.
+
+## Step 1: Create an ArtifactHub repository file
+
+ArtifactHub will need to know that you own the repository you set up for your plugins (this is done in a later step in this guide). So start by creating a file called `artifacthub-repo.yaml` at the root of your GitHub repository with the following content:
+
+```yaml
+owners:
+- name: <Your Name>
+  email: <your-email@example.com>
+```
+
+Replace `<Your Name>` and `<your-email@example.com>` with your name and email, respectively.
+
+## Step 2: Build and Release Your Plugin on GitHub
+
+1. Move to your plugin’s folder, then run the following command:
+  ```bash
+  npm install
+  npm run build
+  healdamp-plugin package
+```
+It will create a tarball of the plugin in the format that is accepted by headlamp and also print the checksum of the tarball that we will use in the next step.
```

Comment:
```suggestion
It will create a tarball of the plugin in a Headlamp-accepted format and also print the checksum of the tarball that we will use in the next step.
```